code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -87,30 +96,38 @@
 						return
 				return
 	
-	if(W.isFlameSource() && fuselength)
-		if(can_light())
-			fuselit = TRUE
-			if(reagents.get_reagent_amount(/datum/reagent/fuel) >= LETHAL_FUEL_CAPACITY && user)
-				msg_admin_attack("[user] ([user.ckey]) lit the fuse on an improvised [name] grenade. (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",ckey=key_name(user))
-			if(fuselength >= 3 && fuselength <= 5 && prob(30) && reagents.get_reagent_amount(/datum/reagent/fuel) >= LETHAL_FUEL_CAPACITY)
-				user.visible_message(SPAN_DANGER("<b>[user]</b> accidentally takes \the [W] too close to \the [name]'s opening!"))
-				detonate(TRUE) // it'd be a bit dull if the toy-levels of fuel had a chance to insta-pop, it's mostly just a way to keep the grenade balance in check
-			if(fuselength in list(1, 2))
-				user.visible_message(SPAN_DANGER("<b>[user]</b> tries to light the fuse on \the [name] but it was too short!"), SPAN_DANGER("You try to light the fuse but it was too short!"))
-				detonate(TRUE) // if you're somehow THAT determined and/or ignorant you managed to get the fuse below 3 seconds, so be it. reap what you sow.
+	else if(W.isFlameSource() && fuselength)
+		light_fuse(W, user)
+	. = ..()
+
+/obj/item/reagent_containers/food/drinks/cans/proc/light_fuse(obj/item/W, mob/user, var/premature=FALSE)
+	if(can_light())
+		fuselit = TRUE
+		update_icon()
+		set_light(2, 2, LIGHT_COLOR_LAVA)
+		if(reagents.get_reagent_amount(/datum/reagent/fuel) >= LETHAL_FUEL_CAPACITY && user)
```

Comment:
Lines 108 and 110 both do the same reagents check. Move it into an earlier if statement.